### PR TITLE
Refactor components to only use AdyenContext.

### PR DESCRIPTION
**Pull request by nauaros**
## Summary
This PR contains the following work items:

- [x] Remove `APIContext` parameter from component initializers.
- [x] Use `apiContext` instance within the `AdyenContext` object.
- [x] Update unit tests.
- [x] Remove `checkoutAttemptId` property from `AnalyticsConfiguration` (**enabled by default**)
- [x] Telemetry analytics enabled by default.
- [x] Finish documentation for `AdyenContext` class.
- [x] Add new classes to jazzy index.

### CODE REVIEW

File: AdyenActions/AdyenActionComponent.swift [hunk 1]
```diff
@@ -15,14 +15,15 @@ import UIKit
  [Implementation Reference](https://github.com/Adyen/adyen-ios#handling-an-action)
  */
 public final class AdyenActionComponent: ActionComponent, ActionHandlingComponent {
-    
-    public let apiContext: APIContext
 
-    /// The Adyen context.
-    public let adyenContext: AdyenContext
-    
+    /// :nodoc:
```
Comment (erenbesel) [inline AdyenActions/AdyenActionComponent.swift:19]: Is it intended to be nodoc?

File: AdyenComponents/Boleto/BoletoComponent.swift [hunk 2]
```diff
@@ -10,12 +10,10 @@ import UIKit
 
 /// A component that provides a form for Boleto payment.
 public final class BoletoComponent: PaymentComponent, LoadingComponent, PresentableComponent, AdyenObserver {
-    
-    /// :nodoc:
-    public let apiContext: APIContext
 
-    /// The Adyen context
-    public let adyenContext: AdyenContext
+    /// :nodoc:
```
Comment (erenbesel) [inline AdyenComponents/Boleto/BoletoComponent.swift:14]: intentional nodoc?

File: AdyenDropIn/Models/DropInConfiguration.swift [hunk 3]
```diff
@@ -36,12 +36,10 @@ public extension DropInComponent {
         
         /// Shopper related information
         public var shopper: PrefilledShopperInformation?
-        
-        /// API context used to retrieve internal resources.
-        public let apiContext: APIContext
 
-        /// The Adyen context.
-        public let adyenContext: AdyenContext
+        /// :nodoc:
```
Comment (erenbesel) [inline AdyenDropIn/Models/DropInConfiguration.swift:40]: intentional nodoc?

File: AdyenActions/AdyenActionComponent.swift (hunk 1 above)
Comment (nauaros) [reply AdyenActions/AdyenActionComponent.swift:19]: Yes! Intended, don't wanna expose this in the docs

File: AdyenComponents/Boleto/BoletoComponent.swift (hunk 2 above)
Comment (nauaros) [reply AdyenComponents/Boleto/BoletoComponent.swift:14]: same

File: AdyenDropIn/Models/DropInConfiguration.swift (hunk 3 above)
Comment (nauaros) [reply AdyenDropIn/Models/DropInConfiguration.swift:40]: Don't show this in the docs

**Review by erenbesel — APPROVED** [review]

**Review by mohammedDehairy — APPROVED** [review]

**Pull request merged**